components/ProfileMenu.vue

```diff
@@ -34,64 +40,94 @@ const logout = async () => {
+                :class="{
+                  'bg-label-separator dark:bg-label-dark-separator': active,
+                }"
+                class="flex w-full items-center justify-between rounded-lg px-3 py-2"
+                @click="
+                  (e: MouseEvent) => {
+                    e.preventDefault();
+                    profileStore.autoplay = !profileStore.autoplay;
+                  }
+                "
               >
-                {{ $t("profile.autoplay") }}
-              </button>
-              <Switch
-                v-model="autoPlayEnabled"
-                :class="autoPlayEnabled ? 'bg-tint' : 'bg-background-2'"
-                class="relative inline-flex h-[24px] w-[40px] shrink-0 cursor-pointer rounded-full p-[4px] transition-colors duration-200 ease-in-out focus:outline-none"
-              >
-                <span
-                  aria-hidden="true"
+                <span>{{ $t("profile.autoplay") }}</span>
+                <Switch
+                  v-model="profileStore.autoplay"
                   :class="
-                    autoPlayEnabled ? 'translate-x-[100%]' : 'translate-x-0'
+                    profileStore.autoplay
+                      ? 'bg-tint dark:bg-tint-dark'
+                      : 'bg-background-2 dark:bg-background-dark-2'
                   "
-                  class="pointer-events-none inline-block h-[16px] w-[16px] transform rounded-full bg-white-1 shadow-lg ring-0 transition duration-200 ease-in-out"
-                />
-              </Switch>
+                  class="relative inline-flex h-6 w-10 shrink-0 cursor-pointer rounded-full p-1 transition-colors duration-200 ease-in-out focus:outline-none"
+                >
+                  <span
+                    aria-hidden="true"
+                    :class="
+                      profileStore.autoplay
+                        ? 'translate-x-full'
+                        : 'translate-x-0'
+                    "
+                    class="pointer-events-none inline-block aspect-square w-4 transform rounded-full bg-white-1 shadow-lg ring-1 ring-black-separator transition duration-200 ease-in-out"
+                  />
+                </Switch>
+              </button>
             </MenuItem>
-            <MenuItem v-slot="{ active }" as="div">
+            <MenuItem v-slot="{ active }">
               <button
-                :class="active ? 'bg-violet-500 text-white' : 'text-gray-900'"
-                class="w-full px-2 py-2 text-left hover:bg-[red]"
+                :class="{
+                  'bg-label-separator dark:bg-label-dark-separator': active,
+                }"
+                class="w-full rounded-lg px-3 py-2 text-left"
                 @click.stop
               >
-                {{ $t("profile.theme") }}
+                <p>{{ $t("profile.theme") }}</p>
+                <span class="text-label-2 dark:text-label-dark-2">
+                  {{ $t(`profile.theme-${$colorMode.value}`) }}
```

Comment:
Please do not use variables for translation text, otherwise our mechanism for detecting unused or missing language snippets (https://eslint-plugin-vue-i18n.intlify.dev/rules/no-unused-keys.html / https://eslint-plugin-vue-i18n.intlify.dev/rules/no-missing-keys.html) won't work.

Reply:
Changed to use a ternary instead of dynamic keys; is that ok enough?